### DISCUSSION TITLE
Why remove reaper() goroutine in pool.go in v9?

### DISCUSSION THREAD

**Discussion by dgqypl** [redis/go-redis#3029] [Q&A]
I read the v8's `pool.go` code and found that there is a reaper() goroutine periodic execute to maintain the `MinIdleConns` connections, so when my business code to execute a redis command, the connection taken from the connection pool is high probability have been created by reaper() goroutine, and it can save the establishing time when the redis command execute.

But the v9.0.1's CHANGELOG only say: "Removed connection reaper in favor of `MaxIdleConns`.", not to explain how to maintain `MinIdleConns` connections. I debug `pool.go` and observe that when a redis command execute, in the `Get(ctx context.Context)` method, first loop do `popIdle()`, and when all connections is unhealthy, then create an idle connection.

So my question is:
1. why remove reaper() goroutine?
2. doesn't it cost more time when execute a redis command if all idle connections are unhealthy at that moment?

**Comment by vmihailenco** [comment]
The main reason is that database/sql pool does not have/require the reaper goroutine. So we tried to adopt a similar approach with Min/MaxIdleConns.

**Reply by dgqypl** [reply]
So my 2nd question is exist? @vmihailenco
